data-prepper-plugins/obfuscate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/obfuscation/ObfuscationProcessorTest.java

```diff
@@ -75,7 +80,23 @@ void setup() {
         lenient().when(mockConfig.getAction()).thenReturn(defaultConfig.getAction());
         lenient().when(mockConfig.getPatterns()).thenReturn(defaultConfig.getPatterns());
         lenient().when(mockConfig.getTarget()).thenReturn(defaultConfig.getTarget());
-        obfuscationProcessor = new ObfuscationProcessor(pluginMetrics, mockConfig, mockFactory);
+        lenient().when(mockConfig.getObfuscateWhen()).thenReturn(null);
+        obfuscationProcessor = new ObfuscationProcessor(pluginMetrics, mockConfig, mockFactory, expressionEvaluator);
+    }
+
+    @Test
+    void obfuscate_when_evaluates_to_false_does_not_modify_event() {
+        final String expression = "/test == success";
+        final Record<Event> record = createRecord(UUID.randomUUID().toString());
+        when(mockConfig.getObfuscateWhen()).thenReturn(expression);
+        when(expressionEvaluator.evaluateConditional(expression, record.getData())).thenReturn(false);
+
+        final ObfuscationProcessor objectUnderTest = new ObfuscationProcessor(pluginMetrics, mockConfig, mockFactory, expressionEvaluator);
+
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) objectUnderTest.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.size(), equalTo(1));
+        assertThat(editedRecords.get(0), equalTo(record));
```

Comment:
I'm not sure this is validating that the event was unmodified.

Ideally, you could verify this with `verifyNoInteractions(record.getData()`, but this `Event` is not a mock. So I think you will need another approach. You should be able to call `record.getData().toMap()` before `doExecute`. 

Then you can assert:

```
assertThat(editedRecords.get(0).getData().toMap(), equalTo(expectedMap));
```